safety-architecture/tools/callgraph-tool/Makefile

```diff
@@ -58,6 +59,20 @@ install-requirements: ## Install python requirements
 	pip3 install -r requirements.txt --no-cache-dir
 	$(call target_success,$@)
 
+install-preprequisites:
+	sudo apt update
+	sudo apt install -y -q python3 \
+		build-essential cmake \
+		clang-10 python3-clang-10 llvm-10 llvm-10-dev clang-format-10
+	@if [ ! -d ${PYENV_DIR} ]; then \
+		printf "${YELLOW}Python env directory does not exist${NC}\n"; \
+		printf "${YELLOW}Creating python env direcotry ${PYENV_DIR}{$NC}\n"; \
+	fi
+	python3 -m venv venv
+	source venv/bin/activate
```

Comment:
I'm not sure I understand the anticipated use of target "install-prerqeuisites". If you source a file from inside a makefile target, I assume it doesn't change the environment outside the target? 

E.g. why is source venv/bin/activate needed here?

Reply:
It is used to automatically setup the working environment prerequisites. Possibly, it could be also used to simplify CI setup.
The whole point of python environment is to contain package installation. If you don't activate it the installation will end up in some global package repository which beats the purpose of the environment.